cli/types/cypress.d.ts

```diff
@@ -2993,6 +2993,7 @@ declare namespace Cypress {
     // Internal or Unlisted at server/lib/config_options
     namespace: string
     projectRoot: string
+    repoRoot: string
     devServerPublicPathRoute: string
```

Comment:
This PR adds `repoRoot` to the Cypress config. We detect it by assuming that the directory with the `.git` folder is the repo root. 

Previously, `relativeFile` in the error codeframe was relative to the Cypress config file. Now, it's relative to the repo root. This fixes an edge case with monorepos where we had an incomplete path for `relativeFile`, resulting in GitHub links to the user's source code being busted in the dashboard.

Question: Our assumption about `.git` being at the repo root might not always be correct -- for example, Tim mentioned that some users may not have git when running in CI, i.e. maybe the test files are hanging out in an S3 bucket. Def an edge case though. **Should we make `repoRoot` a configurable config option? Right now it's internal/unlisted. Would we have to confirm with product?Rachel Ruderman

Reply:
Hmm can you clarify `projectRoot`? It's great to have a format definition for `repoRoot` - might be good to know what exactly the difference it between that and `projectRoot`.

I'd expect `repoRoot` to be nullable - representing a test run where a project doesn't have an Git. Should this be `string | null`? Better yet, `Readonly<string | null>`?

I wonder if it should actually be `gitRepoRoot`? We don't support other source control software (which is why we have `GitDataSource`, not `SourceControlDataSource`, for example).

Is there a scenario a user would want to configure `repoRoot`? I can't think of one.

Reply:
1. Updated to `Readonly<string | null>` 🎉 
2. Re: why not `gitRepoRoot`: it actually started off as `gitRepoRoot` 😄 , but ultimately decided that `git` is just an implementation detail. In the edge case Tim provided where some users won't have git in CI, there are other methods we could use to detect the root, i.e. looking for the root-most `package.json`. However since it's very much an edge case I think it's not the kind of thing we should invest in unless a problem presents itself. Keeping `repoRoot` generic gives us space to expand on the implementation should the need arise. 
3. Re: scenarios where we would want to configure `repoRoot`: in case our auto-detection of repoRoot for whatever reason doesn't work for the user's use case. That's an edge case though, thinking about it now I think it's best to keep things simple -- non-configurable -- unless the need presents itself